src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private void linkLast(T value) {
+        final Node<T> l = last;
+        final Node<T> newNode = new Node<>(l, value, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkBefore(T value, Node<T> succ) {
+        final Node<T> pred = succ.prev;
+        final Node<T> newNode = new Node<>(pred, value, succ);
+        succ.prev = newNode;
+        if (pred == null) {
+            first = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Illegal index: " + index);
+        }
+    }
+
+    private Node<T> node(int index) {
+        checkIndex(index);
```

Comment:
let's also rename this method as well to make it more informative, for example "get(int index)"